src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Controllers/AuthController.cs

```diff
@@ -83,11 +88,49 @@ public async Task SendEmailConfirmLink(SendEmailConfirmLinkRequestDto sendConfir
         var result = await _fluentEmail
             .To(user.Email, user.DisplayName)
             .Subject(EmailStrings.ConfirmationLinkEmailSubject)
-            .UsingTemplateFromEmbedded("TodoTemplate.Api.Resources.EmailConfirmation.cshtml",
-                                    new EmailConfirmationModel
+            .UsingTemplateFromEmbedded($"TodoTemplate.Api.Resources.EmailConfirmation.cshtml",
+                                    new EmailLinksModel
                                     {
                                         DisplayName = user.DisplayName,
-                                        ConfirmationLink = confirmationLink
+                                        Link = confirmationLink
+                                    },
+                                    assembly)
+            .SendAsync(cancellationToken);
+
+        if (!result.Successful)
+            throw new ResourceValidationException(result.ErrorMessages.ToArray());
+    }
+
+    [HttpPost("[action]")]
+    public async Task SendEmailForgotPasswordLink(SendEmailForgotPasswordLinkRequestDto sendConfirmLinkEmailRequest, CancellationToken cancellationToken)
+    {
+        var user = await _userManager.FindByEmailAsync(sendConfirmLinkEmailRequest.Email);
+
+        if (user is null)
+            throw new BadRequestException(nameof(ErrorStrings.UserNameNotFound));
+
+        var token = await _userManager.GeneratePasswordResetTokenAsync(user);
+
+        var tokenBase64 = Convert.ToBase64String(System.Text.Encoding.UTF8.GetBytes(token));
```

Comment:
Do we need base 64 encoding? Why we didn't the same for email confirmation?

Reply:
The problem is '+' sign in the token. It replaces with space so the token became invalid. I encode the token with base64 to solve the issue. Unlike the reset password razor page, there is no problem with the confirmation method in the Auth controller. @ysmoradi

Reply:
I think base64 encoding is overkill here, it just needs an `UrlEncoding`.

Reply:
What's the difference between the reset password token and email confirmation token?
I think both need or don't need URL encoding? @mohabbati

Reply:
I'm pretty sure that both need the url-encoding. any dynamic data that goes to a url need this encoding, you can't be sure about the illegal characters there.

Reply:
> What's the difference between the reset password token and email confirmation token? I think both need or don't need URL encoding? @mohabbati

In the email confirmation, we don't have token so there is no illegal character, but in the reset password, we have.

Reply:
I just encoded token using HttpUtility.UrlEncode() in the link generation. The point is that in the razor partial class it does not need to decode. It seems asp .net decodes it automatically. Is it enough? @msynk @ysmoradi

Reply:
Maybe, what do you think? @msynk 
Meanwhile, could you please test your theory with an illegal character? @mohabbati

Reply:
I tested it with illegal characters and it worked well. @ysmoradi

Reply:
yeah, the decoding part is done automatically by the framework itself.
great job guys :)